packages/fxa-auth-server/lib/payments/configuration/base.ts

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import joi from '@hapi/joi';
+
+export type CapabilityConfig = {
+  [key: string]: string;
+};
+
+const capabilitySchema = joi
+  .object()
+  .pattern(joi.string(), joi.string().min(1))
+  .min(1);
+
+export type UrlConfig = {
+  download?: string;
+  webIcon?: string;
+  emailIcon?: string;
+  termsOfService?: string;
+  termsOfServiceDownload?: string;
+  privacyNotice?: string;
+  privacyNoticeDownload?: string;
+  playStore?: string;
+  appStore?: string;
+};
+
+const urlsSchema = joi.object({
+  download: joi.string().uri(),
+  webIcon: joi.string().uri(),
+  emailIcon: joi.string().uri(),
+  termsOfService: joi.string().uri(),
+  termsOfServiceDownload: joi.string().uri(),
+  privacyNotice: joi.string().uri(),
+  privacyNoticeDownload: joi.string().uri(),
+  playStore: joi.string().uri(),
+  appStore: joi.string().uri(),
+});
+
+export type UiContentConfig = {
+  subtitle?: string;
+  details?: string[];
+  successActionButtonLabel?: string;
+  upgradeCTA?: string;
+};
+
+const uiContentSchema = joi.object({
+  subtitle: joi.string(),
+  details: joi.array().items(joi.string()),
+  successActionButtonLabel: joi.string(),
+  upgradeCTA: joi.string(),
+});
+
+export type StyleConfig = {
+  webIconBackground?: string;
+};
+
+const stylesSchema = joi.object({
+  webIconBackground: joi.string().required(),
+});
+
+export type SupportConfig = {
+  app: string[];
+};
+
+const supportSchema = joi.object({
+  app: joi.array().items(joi.string()),
+});
+
+/**
+ * Base config class that has common components both Products and Plans
+ * can extend from.
+ */
+export interface BaseConfig {
+  active: boolean;
+  capabilities?: CapabilityConfig;
+  urls?: UrlConfig;
```

Comment:
I see that you're using this as the base for both product and plans, but how do we enforce that they are required on at least the product or plan?